protocols/kad/src/dht.proto

```diff
@@ -26,6 +26,9 @@ message Record {
     // The remaining TTL of the record, in seconds.
     // Currently specific to rust-libp2p.
     uint32 ttl = 777;
+
+	// Client mode or server mode.
+	bool client = false;
```

Comment:
The fact that a node is in client or server mode is never transfered over the wire. Instead, one signals to be in server mode by offering the `/libp2p/kad/0.1.0` protocol on new substreams and via the identify protocol.
```suggestion
```